src/resolvers/InstallableAppResourceResolver.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { ISubscriptionContext } from "@microsoft/vscode-azext-utils";
+import { AppResource, AppResourceResolver, ResolvedAppResourceBase } from "../api";
+import { getAzureExtensions } from "../AzExtWrapper";
+
+/**
+ * This resolver acts as a "placeholder" resolver when an extension is known for the resource type but not installed (or not enabled).
+ * Upon the resolved node being clicked, it will open the extension page on the marketplace to allow them to easily install (or enable).
+ */
+class InstallableAppResourceResolver implements AppResourceResolver {
+    public resolveResource(_subContext: ISubscriptionContext, resource: AppResource): ResolvedAppResourceBase {
+        // We know the extension is known but uninstalled, or else it would not have passed the `isApplicable` check below
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const extensionId = getAzureExtensions().find(azExt => azExt.matchesResourceType(resource))!.id;
+
+        return {

```

Comment:
I think we can explore using the vscode.command to just install the extension instead of taking them to the marketplace page.

Reply:
I'd like to hear from @fiveisprime on that too; personally I lean toward showing them the marketplace page rather than auto-installing as a sort of "good faith" way of making sure they know exactly what they're doing. Or you could call it C.Y.A. :smile:

Reply:
Yeah. As much as I'd love to auto-install, it just doesn't quite feel right.

Reply:
Yeah it shows this modal after running the `workbench.extensions.installExtension` command.
<img width="552" alt="image" src="https://user-images.githubusercontent.com/12476526/157756220-e19028d7-2cd5-47c3-bd66-dc72ebfab62a.png">

Reply:
Oh nice, I dig that! :shipit:

Reply:
That's neat! I'll make that change. I agree, long as we are giving them this prompt I'm happy.

Reply:
@nturinski @fiveisprime @alexweininger I ran into one problem with the `workbench.extensions.installExtension`. If the extension is _installed_ but _disabled_, when you execute the command nothing happens. As far as I know it is not possible for us to tell the difference between uninstalled and disabled--either way the VSCode extension API returns `undefined`. I also do not see any commands for enabling extensions.

I don't know how common of a problem we expect this to be, but it is a point in favor of just opening the marketplace page.

Reply:
Ok, I think we should talk to Isidor about providing a better API for prompting to install extensions. But for now, taking users to the marketplace page seems like the safe bet.